### PR TITLE
Add event proto

### PR DESCRIPTION
**Pull request by alexmasi**
This will be used in anonymous usage reporting

### CODE REVIEW

File: proto/event.proto [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+message Topology {
+  repeated Node nodes = 1;
+  int64 link_count = 2;
+}
+
+message Node {
+  topo.Vendor vendor = 1;
+  string model = 2;
```
Comment (bormanp) [inline proto/event.proto:85]: I could see some objection to this when the model of the device is unreleased.   I think tags 2 through 4 probably should be optional.

File: proto/event.proto [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  string version = 3;
```
Comment (bormanp) [inline proto/event.proto:86]: How important are version and os?  Wouldn't these be determinable from the vendor/model?  There probably should be a comment justifying each piece of information gathered.

**Review by bormanp — CHANGES_REQUESTED** [review]
I think there should be a comment that says identifying information must not be added to these structures (which is why link_count is included but the list of links is not).

File: proto/event.proto (hunk 2 above)
Comment (alexmasi) [reply proto/event.proto]: Added comments. Version and OS provide much less value than knowing the vendor+model so I've removed them.

File: proto/event.proto (hunk 1 above)
Comment (alexmasi) [reply proto/event.proto:107]: We can configure this through the KNE config file. By default all reporting will be set to false. Later we can added an enum in the cfg to fine tune what is shared

**Review by bormanp — APPROVED** [review]

**Pull request merged**